File: Components/WinForms/VB/ApplicationUIMainDemo/Modules/WidgetView/MaximizedWidgetControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace DevExpress.ApplicationUI.Demos
	Partial Public Class MaximizedWidgetControl
		Inherits UserControl
		Private updateTimer As Timer = New Timer()
		Private index As Integer = 0
		Private r As Random = New Random()
		Public Sub New()
			InitializeComponent()
			updateTimer.Interval = 1000
			AddHandler updateTimer.Tick, AddressOf OnTimerTick
			updateTimer.Start()
			For i As Integer = 0 To 29
				AddNewSeriesPoint()
			Next i
		End Sub
		Private Const BeginDate As Integer = 40000
		Protected Sub AddNewSeriesPoint()
			Dim data As StockData = StockDataCreator.GetData(index)
			If IsDisposed Then
				updateTimer.Stop()
				Return
			End If
			Dim seriesPoint1 As SeriesPoint = New SeriesPoint(data.Date, New Object() { CObj(data.LowPrice), CObj(data.HighPrice), CObj(data.OpenPrice), CObj(data.ClosePrice) })
			chartControl1.Series(0).Points.Add(seriesPoint1)
			Dim [date] As Integer = BeginDate
			If index > 40 Then
				[date] = BeginDate + index - 30
			End If
			TryCast(chartControl1.Diagram, IXYDiagram2D).AxisX.WholeRange.SetMinMaxValues(DateTime.FromOADate([date]), data.Date)
			TryCast(chartControl2.Diagram, IXYDiagram2D).AxisX.WholeRange.SetMinMaxValues(DateTime.FromOADate([date]), data.Date)
			index += 1
			Dim seriesPoint2 As SeriesPoint = New SeriesPoint(data.Date, data.Volumne)
			chartControl2.Series(0).Points.Add(seriesPoint2)
		End Sub
		Private Sub OnTimerTick(ByVal sender As Object, ByVal e As EventArgs)
			AddNewSeriesPoint()
		End Sub
	End Class
End Namespace

File: Components/WinForms/CS/ApplicationUIMainDemo/Modules/WidgetView/MaximizedWidgetControl.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace DevExpress.ApplicationUI.Demos {
    public partial class MaximizedWidgetControl : UserControl {
        Timer updateTimer = new Timer();
        int index = 0;
        Random r = new Random();
        public MaximizedWidgetControl() {
            InitializeComponent();
            if(components != null)
                components.Add(updateTimer);
            updateTimer.Interval = 1000;
            updateTimer.Tick += OnTimerTick;
            updateTimer.Start();
            for(int i = 0; i < 30; i++)
                AddNewSeriesPoint();
        }
        void OnTimerTick(object sender, EventArgs e) {
            AddNewSeriesPoint();
        }
        const int BeginDate = 40000;
        protected void AddNewSeriesPoint() {
            if(IsDisposed) {
                updateTimer.Stop(); 
                return;
            }
            chartControl1.BeginInit();
            chartControl2.BeginInit();
            StockData data = StockDataCreator.GetData(index);
            var points1 = chartControl1.Series[0].Points;
            SeriesPoint seriesPoint1 = new SeriesPoint(data.Date, (object)data.LowPrice, (object)data.HighPrice, (object)data.OpenPrice, (object)data.ClosePrice);
            points1.Add(seriesPoint1);
            int date = BeginDate;
            if(index > 40)
                date = BeginDate + index - 30;
            (chartControl1.Diagram as IXYDiagram2D).AxisX.WholeRange.SetMinMaxValues(DateTime.FromOADate(date), data.Date);
            (chartControl2.Diagram as IXYDiagram2D).AxisX.WholeRange.SetMinMaxValues(DateTime.FromOADate(date), data.Date);
            index++;
            var points2 = chartControl2.Series[0].Points;
            SeriesPoint seriesPoint2 = new SeriesPoint(data.Date, data.Volumne);
            points2.Add(seriesPoint2);
            Cleanup(points1, points2);
            chartControl1.EndInit();
            chartControl2.EndInit();
        }
        void Cleanup(SeriesPointCollection points1, SeriesPointCollection points2) {
            if(index > 40) {
                points1.RemoveAt(0);
                points2.RemoveAt(0);
            }
        }
    }
}
